zaza/openstack/charm_tests/ceph/tests.py

```diff
@@ -690,28 +702,162 @@ def expected_apps(self):
-            sync_states.append(sync_check_str in stdout)
+            sync_states.append((data_check in stdout) and
+                               (meta_check in stdout))
         assert all(sync_states)
 
-    def test_processes(self):
+    @tenacity.retry(wait=tenacity.wait_exponential(multiplier=1, max=60),
+                    reraise=True, stop=tenacity.stop_after_attempt(12))
+    def fetch_rgw_object(self, target_client, container_name, object_name):
+        '''
+        Fetch RGW object content
+        :param target_client: boto3 client object configured for an endpoint.
+        :param container_name: RGW bucket name for desired object.
+        :param object_name: Object name for desired object.
+        '''
+        return target_client.Object(container_name, object_name).get()[
+            'Body'
+        ].read().decode('UTF-8')
+
+    def promote_rgw_to_primary(self, app_name:str):
+        '''Promote provided app to Primary and update period at new secondary'''
+        if app_name is self.primary_rgw_app:
+            new_secondary = self.secondary_rgw_unit
+        else:
+            new_secondary = self.primary_rgw_unit
+
+        # Promote to Primary
+        zaza_model.run_action_on_leader(
+            app_name,
+            'promote',
+            action_params={},
+        )
+
+        # Period Update Commit new secondary.
+        cmd = self.get_rgwadmin_cmd_skeleton(new_secondary)
+        zaza_model.run_on_unit(
+            new_secondary, cmd + 'period update --commit'
+        ).get('Stdout', '')
+
+    def get_client_keys(self, rgw_app_name=None):
+        """Create access_key and secret_key for boto3 client"""
+        unit_name = self.primary_rgw_unit
+        if rgw_app_name is not None:
+            unit_name = rgw_app_name + '/0'
+        user_name = 'botoclient'
+        cmd = self.get_rgwadmin_cmd_skeleton(unit_name)
+        users = json.loads(zaza_model.run_on_unit(
+            unit_name, cmd + 'user list'
+        ).get('Stdout', ''))
+        # Fetch boto3 user keys if user exists.
+        if user_name in users:
+            output = json.loads(zaza_model.run_on_unit(
+                unit_name, cmd + 'user info --uid={}'.format(user_name)
+            ).get('Stdout', ''))
+            keys = output['keys'][0]
+            return keys['access_key'], keys['secret_key']
+        # Create boto3 user if it does not exist.
+        create_cmd = cmd + 'user create --uid={} --display-name={}'.format(
+            user_name, user_name
+        )
+        output = json.loads(
+            zaza_model.run_on_unit(unit_name, create_cmd).get('Stdout', '')
+        )
+        keys = output['keys'][0]
+        return keys['access_key'], keys['secret_key']
+
+    @tenacity.retry(
+        retry=tenacity.retry_if_result(lambda ret: ret is None),
+        wait=tenacity.wait_fixed(10),
+        stop=tenacity.stop_after_attempt(5)
+    )
+    def get_rgw_endpoint(self, unit_name:str, isHttps=False):
```

Comment:
We generally use snake_case rather than lowerCamelCase for variable names (ref: `isHttps`).

I also wonder if we can tell if TLS is enabled by checking for Vault inside this function

Reply:
Boolean parameters look a little strange in snake case (isHttps vs is_https). However the better solution is to just check for vault and decide on endpoint, I have shifted to that approach.